### ISSUE TITLE
Difference between the masks generated by Meta-demo v/s local

### ISSUE THREAD

**Issue by bsinghpratap** [facebookresearch/segment-anything#330]
Masks generated by on the demo ![image](https://user-images.githubusercontent.com/7297516/236582762-10ab3c00-5b73-4f63-ab9e-dfe0dd3143fd.png) through the online demo (https://segment-anything.com/demo) are different from the masks generated in the example notebook (https://github.com/facebookresearch/segment-anything/blob/main/notebooks/automatic_mask_generator_example.ipynb). 
- I noticed that the online demo performs some de-duplication of masks. Is that the reason the masks generated are different? 
- How can I achieve the same masks as the online demo? Would I have to tune/change some of the hyper-parameters?

**Comment by bsinghpratap** [comment]
![Masks generated through the everything functionality of demo](https://user-images.githubusercontent.com/7297516/236583185-c7d95373-f18f-4b0a-aafc-95e8feacade1.png)

**Comment by githublqs** [comment]
i have the same question！[https://github.com/facebookresearch/segment-anything/issues/346](url)

**Comment by githublqs** [comment]
I have the same question  #346

**Comment by anlarro** [comment]
Any update on this? The results I get using the online demo are really good but can't get the same results with the mask_generator.

**Comment by GuaGod** [comment]
image
You can try to tune some parameters in the SamAutomaticMaskGenerator, such as pred_iou_thresh and stalility_score_thresh.And there are some improvement when I set lower value at them.

**Comment by GuaGod** [comment]
> Any update on this? The results I get using the online demo are really good but can't get the same results with the mask_generator.

![image](https://github.com/facebookresearch/segment-anything/assets/39557078/24e4f999-533c-4f47-ae6d-d04778227723)
You can try to tune some parameters in the SamAutomaticMaskGenerator, such as pred_iou_thresh and stability_score_thresh.And I get better results after set lower values at them.

**Comment by anlarro** [comment]
> > Any update on this? The results I get using the online demo are really good but can't get the same results with the mask_generator.
> 
> ![image](https://user-images.githubusercontent.com/39557078/243733889-24e4f999-533c-4f47-ae6d-d04778227723.png) You can try to tune some parameters in the SamAutomaticMaskGenerator, such as pred_iou_thresh and stability_score_thresh.And I get better results after set lower values at them.

Thanks. I tried many different parameters already but nothing similar to what I get with the online demo. It would be nice if somebody can provide the parameters used for the online demo.

**Comment by GuaGod** [comment]
Yeah.I think the largest model is the best model in the all three local models.But there is a performance gap bewteen it and the online model.


发自我的iPhone

------------------ Original ------------------
From: Andrés Larroza ***@***.***&gt;
Date: 周二,6月 6,2023 11:46 下午
To: facebookresearch/segment-anything ***@***.***&gt;
Cc: Liangjie ***@***.***&gt;, Manual ***@***.***&gt;
Subject: Re: [facebookresearch/segment-anything] Difference between the masks generated by Meta-demo v/s local (Issue #330)





   
Any update on this? The results I get using the online demo are really good but can't get the same results with the mask_generator.
  
 You can try to tune some parameters in the SamAutomaticMaskGenerator, such as pred_iou_thresh and stability_score_thresh.And I get better results after set lower values at them.
  
Thanks. I tried many different parameters already but nothing similar to what I get with the online demo. It would be nice if somebody can provide the parameters used for the online demo.

**Comment by Mr-HanSansui** [comment]
I met the same question, did anyone solve this problem?

**Comment by BalajiAI** [comment]
Even, I'm also facing the same issue. The Online demo is giving good results, whereas the colab notebook isn't.
(Will update if I found any solutions.)

**Comment by Nivratti** [comment]
The primary reason for the variation in masks between the online demo and the local notebook is that the online demo uses a different model for segment everything functionality.  The online demo is powered by the model hosted at the following endpoint:

[segment_everything_box_model](https://model-zoo.metademolab.com/predictions/segment_everything_box_model)
